sal/tests/client/unit/test_auth.py

```diff
@@ -0,0 +1,284 @@
+                             token,
+                             auth_args=[UN, PW])
+
+
+def test_authenticate_prompt_password(patched_client_with_auth,
+                                      server_auth_response,
+                                      host,
+                                      token,
+                                      patch_get):
+
+    """
+    GIVEN
+        A client with a host URL
+        AND a username passed to the client
+    WHEN
+        The client authenticates with the host
+    THEN
+        The user is prompted for a password
+        AND the client stores a token received from the host
+    """
+
+    with patch_get as requests_get:
+        with patch('sal.client.main.getpass.getpass', return_value=PW) as gp:
+            check_authentication(host,
+                                 patched_client_with_auth,
+                                 requests_get,
+                                 token,
+                                 auth_args=[UN])
+            
+            gp.assert_called()
+
+
+def test_authenticate_prompt_credentials(patched_client_with_auth,
+                                         server_auth_response,
+                                         host,
+                                         token,
+                                         patch_get):
+
+    """
+    GIVEN
+        A client with a host URL
+    WHEN
+        The client authenticates with the host
+    THEN
+        The user is prompted for a username
+        AND the user is prompted for a password
+        AND the client stores a token received from the host
+    """
+
+    with patch_get as requests_get:
+        with patch('sal.client.main.getpass.getpass', return_value=PW) as gp:
+            with patch('sal.client.main.input',return_value=UN) as inpt:
+                check_authentication(host,
+                                     patched_client_with_auth,
+                                     requests_get,
+                                     token,
+                                     auth_args=[])
+            
+                inpt.assert_called()
+                gp.assert_called()
+
+
+def test_authenticate_default_credentials_file(patched_client_with_auth,
+                                               server_auth_response,
+                                               host,
+                                               token,
+                                               patch_get):
+
+    """
+    GIVEN
+        A client with credentials prompting disabled
+        AND a credentials file in the default location with the format:
+            [host_url]
+            user = username
+            password = password
+    WHEN
+        The client attempts to authenticate
+    THEN
+        A request is sent using credentials from the credentials file 
+    """
+
+    cred_file = {'https://sal.testing': {'user': UN,
+                                        'password': PW}}
```

Comment:
**suggestion:** As in `conftest.py`, the host address is hard coded; the fixture could be used instead.

Reply:
Thanks, changed all three.